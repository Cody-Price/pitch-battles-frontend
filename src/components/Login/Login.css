.login-form {
  position: absolute;
  padding-top: 1rem;
  width: 400px;
  height: 484px;
  top: 0;
  right: 300px;
  transition: 0.3s;
}

.login-form.true {
  right: 0;
}

.login-form {
  padding-top: 0.5rem;

  position: absolute;
  width: 300px;
  right: 300px;
  margin: 1.2rem auto;
  transition: 0.3s;
  top: 0;
  font-family: "chronotype";
  font-size: 1.3rem;
}

.login-form-label {
  margin-top: 0;
}

.signup-form.true {
  left: 0px;
}

.login-form input {
  display: block;
  margin: 0.7rem auto;
  font-family: "chronotype";
  font-size: 1.1rem;
  padding-left: 1.6rem;
  padding-top: 4px;
}

.role-label {
  font-size: 1.1rem;

  margin: 1rem 0 0.4rem;
}

.login-form-button,
.signup-form-switch-button {
  margin: 0.6rem 0;
  font-family: "chronotype";
  font-size: 1.1rem;
  padding-top: 4px;
  border-radius: 1px;
}

.signup-form-switch-button {
  margin: 8rem 0 0 0;
}

.login-form button:hover {
  cursor: pointer;
}

.login-form-form {
  position: relative;
}

.login-email-icon {
  position: absolute;
  background: url(../Landing/images/mail.png);
  width: 17px;
  height: 17px;
  background-size: cover;
  image-rendering: pixelated;
  left: 56px;
  top: 3px;
}

.login-pw-icon {
  position: absolute;
  background: url(../Landing/images/lock.png);
  width: 13px;
  height: 16px;
  background-size: cover;
  image-rendering: pixelated;
  left: 58px;
  top: 39px;
}
