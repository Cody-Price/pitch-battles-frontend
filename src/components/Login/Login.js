import React, { Component } from "react";
import "./Login.css";

class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: "",
      password: ""
    };
  }

  handleChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  render() {
    return (
      <section className={`login-form ${this.props.status}`}>
        <form>
          <input
            placeholder="email"
            type="email"
            name="email"
            onChange={this.handleChange}
            value={this.state.email}
          />
          <input
            placeholder="password"
            type="password"
            name="password"
            onChange={this.handleChange}
            value={this.state.password}
          />
          <button>Login</button>
        </form>
        <button onClick={this.props.landingChange}>Sign Up</button>
      </section>
    );
  }
}

export default Login;
