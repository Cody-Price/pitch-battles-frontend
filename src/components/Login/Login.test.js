import { shallow } from "enzyme";
import React from "react";
import Login from "./Login";

const Enzyme = require("enzyme");
const EnzymeAdapter = require("enzyme-adapter-react-16");

Enzyme.configure({ adapter: new EnzymeAdapter() });

jest.mock("../../utilities/fetchCalls");

describe("Login", () => {
  let wrapper;
  let mockLoginUser;

  beforeEach(() => {
    mockLoginUser = jest.fn();

    wrapper = shallow(<Login loginUser={mockLoginUser} />);
  });

  it("should exist", () => {
    expect(wrapper).toBeDefined();
  });

  it("should render like snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("handleChange", () => {
    it("should set state on handleChange", () => {
      const event = {
        target: {
          name: "email",
          value: "haleyljacobs@gmail.com"
        }
      };

      wrapper.instance().handleChange(event);
      expect(wrapper.state().email).toEqual("haleyljacobs@gmail.com");
    });

    it("should call handleChange on change - email", () => {
      const mockEvent = {
        target: {
          value: "hello",
          name: "email"
        }
      };

      const spy = jest.spyOn(wrapper.instance(), "handleChange");

      wrapper.instance().forceUpdate();

      wrapper.find(".login-email-input").simulate("change", mockEvent);

      expect(spy).toHaveBeenCalledWith(mockEvent);
    });

    it("should call handleChange on change - password", () => {
      const mockEvent = {
        target: {
          value: "hello",
          name: "password"
        }
      };

      const spy = jest.spyOn(wrapper.instance(), "handleChange");

      wrapper.instance().forceUpdate();

      wrapper.find(".login-password-input").simulate("change", mockEvent);

      expect(spy).toHaveBeenCalledWith(mockEvent);
    });
  });

  describe("handleSubmit", () => {
    it("should call preventDefault", () => {
      const mockPreventDefault = jest.fn();

      const mockEvent = {
        preventDefault: mockPreventDefault
      };

      wrapper.instance().handleSubmit(mockEvent);

      expect(mockPreventDefault).toHaveBeenCalled();
    });

    it("should call loginUser with the correct params", () => {
      const mockState = {
        email: "test@mail.com",
        password: "test-password"
      };

      const mockPreventDefault = jest.fn();

      const mockEvent = {
        preventDefault: mockPreventDefault
      };

      wrapper.setState({
        email: mockState.email,
        password: mockState.password
      });

      wrapper.instance().handleSubmit(mockEvent);

      expect(mockLoginUser).toHaveBeenCalledWith(mockState);
    });

    it("should call handleSubmit on submit", () => {
      const spy = jest.spyOn(wrapper.instance(), "handleSubmit");
      const mockPreventDefault = jest.fn();

      const mockEvent = {
        preventDefault: mockPreventDefault
      };

      wrapper.instance().forceUpdate();

      wrapper.find(".login-form-form").simulate("submit", mockEvent);

      expect(spy).toHaveBeenCalledWith(mockEvent);
    });
  });
});
