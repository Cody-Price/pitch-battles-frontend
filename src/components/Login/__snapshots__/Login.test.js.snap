// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form>
          <p>
            email
          </p>
          <input
            name="email"
            onChange={[Function]}
            type="email"
            value=""
          />
          <p>
            password
          </p>
          <input
            name="password"
            onChange={[Function]}
            type="password"
            value=""
          />
          <button>
            Login
          </button>
        </form>,
        <button>
          Sign Up
        </button>,
      ],
      "className": "login-form undefined",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              email
            </p>,
            <input
              name="email"
              onChange={[Function]}
              type="email"
              value=""
            />,
            <p>
              password
            </p>,
            <input
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            />,
            <button>
              Login
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "email",
            },
            "ref": null,
            "rendered": "email",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "password",
            },
            "ref": null,
            "rendered": "password",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
            },
            "ref": null,
            "rendered": "Login",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Up",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Sign Up",
        "type": "button",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form>
            <p>
              email
            </p>
            <input
              name="email"
              onChange={[Function]}
              type="email"
              value=""
            />
            <p>
              password
            </p>
            <input
              name="password"
              onChange={[Function]}
              type="password"
              value=""
            />
            <button>
              Login
            </button>
          </form>,
          <button>
            Sign Up
          </button>,
        ],
        "className": "login-form undefined",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                email
              </p>,
              <input
                name="email"
                onChange={[Function]}
                type="email"
                value=""
              />,
              <p>
                password
              </p>,
              <input
                name="password"
                onChange={[Function]}
                type="password"
                value=""
              />,
              <button>
                Login
              </button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "email",
              },
              "ref": null,
              "rendered": "email",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "password",
              },
              "ref": null,
              "rendered": "password",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
              },
              "ref": null,
              "rendered": "Login",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "button",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
